docs/quickstart/java.md

```diff
@@ -1,33 +1,147 @@
 ---
-title: Java Quickstart
-headline: Java Quickstart
+title: Getting Started with Spine in Java
+headline: Getting Started with Spine in Java
 bodyclass: docs
 layout: docs
 sidenav: doc-side-quickstart-nav.html
 type: markdown
 ---
-
-<p>This guide gets you started with minimal server-side application in Java with a simple
+<p>This guide helps you get started with minimal server-side application in Java with a simple
 working example.</p>
+<hr>
+## Prerequisites
+1.  Make sure you have JDK 8 or higher installed.
+2.  Clone the following source code into your Git repository: 
+  ```bash
+  git clone git@github.com:SpineEventEngine/server-quickstart.git
+  ```
+3. Run `./gradlew clean build` (or `gradlew.bat clean build` on Windows) in the project root folder.
+ 
+## Project Structure
+ 
+The project consists of the following three modules: 
+ * the `model` module
+ * the `server` module
+ * the `client` module
+ 
+Their definitions and the process of their creation is described below. 
+ 
+### The `model` Module
+ 
+The `model` module defines  [Ubiquitous Language](https://martinfowler.com/bliki/UbiquitousLanguage.html) 
+ of the application in Protobuf.
+ 
+The `model/src/main/proto` directory contains the Protobuf definitions of the domain model:
+* `Task` is an aggregate state type; as any entity type, it is marked with the `(entity)` option;
+* `TaskCreated` in `events.proto` is an event of the `TaskAggregate`;
+* `CreateTask` in `commands.proto` is a command handled by the `TaskAggregate`;
```

Comment:
Put this before describing the event. Commands usually come first. This command does come before the event.